### PR TITLE
Mapnik fonts

### PR DESCRIPTION
**Pull request by umeier**
Allow custom fonts for mapnik.
Should solve issue #69

### CODE REVIEW

File: doc/sources.rst [hunk 1]
```diff
@@ -526,6 +526,11 @@ Set to ``true`` to render from mapnik sources with background-color="transparent
 Set the `Mapnik scale_factor <https://github.com/mapnik/mapnik/wiki/Scale-factor>`_ option. Mapnik scales most style options like the width of lines and font sizes by this factor.
 See also :ref:`hq_tiles`.
 
+``fonts``
```
Comment (olt) [inline doc/sources.rst]: Can you rename this to ``font_dir``. This is in line with the `globals.image.font_dir` option used for the attributions.

File: mapproxy/source/mapnik.py [hunk 2]
```diff
@@ -118,6 +124,9 @@ def render_mapfile(self, mapfile, query):
     def _render_mapfile(self, mapfile, query):
         start_time = time.time()
 
+        if self.fonts:
+            mapnik.register_fonts(self.fonts)
```
Comment (olt) [inline mapproxy/source/mapnik.py]: Can you move this to `__init__`? It should be enough to register each directory once.

File: mapproxy/test/system/test_mapnik.py [hunk 3]
```diff
@@ -57,8 +57,28 @@
...
+        <StyleName>marker</StyleName>
+        <Datasource>
+            <Parameter name="type">ogr</Parameter>
+            <Parameter name="file">test_point.geojson</Parameter>
+            <Parameter name="layer">OGRGeoJSON</Parameter>
+        </Datasource>
+    </Layer>
+    <Style name="marker">
+        <Rule>
+            <TextSymbolizer face-name="Fira Sans Regular">[name]</TextSymbolizer>
```
Comment (olt) [inline mapproxy/test/system/test_mapnik.py]: We already bundle a font for the attribution lines in `mapproxy/image/fonts`. Can you use this one?

File: mapproxy/test/system/test_mapnik.py [hunk 4]
```diff
@@ -77,19 +97,24 @@ def setup_module():
         f.write(mapnik_xml)
     with open(os.path.join(test_config['base_dir'], 'mapnik-transparent.xml'), 'wb') as f:
         f.write(mapnik_transp_xml)
+    with open(os.path.join(test_config['base_dir'], 'mapnik-fonts.xml'), 'wb') as f:
+        f.write(mapnik_fonts_xml)
+    test_config['fonts_dir'] = os.path.join(test_config['base_dir'], 'fonts')
+    shutil.copytree(os.path.join(test_config['fixture_dir'], 'fonts'), test_config['fonts_dir'])
```
Comment (olt) [inline mapproxy/test/system/test_mapnik.py]: See comment above.

**Comment by umeier** [comment]
Short of time at the moment. Will do the requested changes around mid of june. Sorry for the delay.

**Comment by weskamm** [comment]
Closing this due to inactivity. Feel free to reopen if you address the remaining issues

**Pull request closed without merging**